### PR TITLE
[kf5] recover from #4652

### PR DESCRIPTION
**Pull request by cenit**
as per title, recovering #4652 (will need a big merge from master...)

### CODE REVIEW

**Comment by cenit** [comment]
due to broken dependencies in master, the CI cannot really test almost any addition.
Work here is suspended until proper fixes land on master

**Comment by Voskrese** [comment]
kf5i18n failed
reason, installed GCC

ninja why it considers that the need to use GCC 
instead MSVC

```
vcpkg_acquire_msys(MSYS_ROOT PACKAGES mingw-w64-i686-gettext)
set(GETTEXT_PATH ${MSYS_ROOT}/mingw32/bin)
vcpkg_add_to_path(${GETTEXT_PATH})
```

decided to this dramatically

```
vcpkg_find_acquire_program(XGETTEXT)
get_filename_component(XGETTEXT_DIR "${XGETTEXT}" DIRECTORY)
set(ENV{PATH} ";$ENV{PATH};${XGETTEXT_DIR}")
```

vcpkg_find_acquire_program.cmake

```
  elseif(VAR MATCHES "XGETTEXT")
    set(PROGNAME xgettext)
    set(PATHS "${DOWNLOADS}/tools/xgettext/bin")
    set(URL "https://github.com/mlocati/gettext-iconv-windows/releases/download/v0.20.1-v1.16/gettext0.20.1-iconv1.16-static-32.zip")
    set(ARCHIVE "gettext0.20.1.zip")
    set(HASH 9b423bc72c06ce8e1f59842a3ec2d8e2ccab2179179f733a83703d4145300ca9053758ca1232245435e88a82983a785c3b51030eea4a5aaa7c3d7d3995b1c040)	
```

**Comment by Voskrese** [comment]
kf5service:x64-windows failed

>   has asked CMake to find a package configuration file provided by
>   "KF5DocTools", but CMake did not find one.

**Comment by JackBoosY** [comment]
Any progress?

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
Thanks for the PR!

Could you please remove 'include(vcpkg_common_functions)' in all portfile.cmake, and add homepage to CONTROL file? Others looks good for me.

File: ports/editorconfig-core-c/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,42 @@
+include(vcpkg_common_functions)
+
+if(VCPKG_TARGET_IS_UWP)
+    message(FATAL_ERROR "${PORT} does not currently support UWP")
+endif()
```
Comment (PhoebeHui) [inline ports/editorconfig-core-c/portfile.cmake]: Could you please use 'vcpkg_fail_port_install(ON_TARGET "UWP")' instead?

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
The CI failed with '##[error] REGRESSION: editorconfig-core-c:x64-uwp', from your changes, this is expected, could you add it to https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt?

**Comment by cenit** [comment]
merged with master to solve conficts. But unfortunately I cannot work on this PR apart from this small maintenance. Happy to transfer it to whomever wants to continue

**Comment by JackBoosY** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Comment by lijingjian001** [comment]
Anyone could tell me how to and kf5 other ports to vcpg localy? where is the documation?

**Pull request closed without merging**